src/transform/WidenImage.js

```diff
@@ -1,30 +1,99 @@
 import './WidenImage.css'
 import elementUtilities from './ElementUtilities'
 
+/**
+ * Gets array of ancestors of element which need widening.
+ * @param  {!HTMLElement} element
+ * @return {!Array.<HTMLElement>} Zero length array is returned if no elements should be widened.
+ */
+const ancestorsToWiden = element => {
+  const widenThese = []
+  let el = element
+  while (el.parentNode) {
+    el = el.parentNode
+    // No need to walk above 'content_block'.
+    if (el.classList.contains('content_block')) {
+      break
+    }
+    widenThese.push(el)
+  }
+  return widenThese
+}
+
+/**
+ * Sets style value.
+ * @param {HTMLElement.style} style
+ * @param {!string} key
+ * @param {*} value
+ * @return {void}
+ */
+const updateStyleValue = (style, key, value) => {
+  style[key] = value
+}
+
+/**
+ * Sets style value only if value for given key already exists.
+ * @param {HTMLElement.style} style
+ * @param {!string} key
+ * @param {*} value
+ * @return {void}
+ */
+const updateExistingStyleValue = (style, key, value) => {
+  const existingValue = style[key]
+  if (!existingValue) {
+    return
+  }
+  updateStyleValue(style, key, value)
+}
+
+/**
+ * Array of arrays of image widening CSS key/value pairs.
+ * @type {Array}
+ */
+const styleWideningKeysAndValues = [
+  ['width', '100%'],
+  ['height', 'auto'],
+  ['maxWidth', '100%'],
+  ['float', 'none']
+]
+
+/**
```

Comment:
It's a matter of opinion but I think the previous implementation was easier to read because it was a dead simple 30 lines. In my estimation, widenElementByUpdatingStyles and widenElementByUpdatingStyles don't offer much value because they're each only used in one place and are their bodies are only two lines. Similarly, updateExistingStyleValue and updateStyleValue could be replaced with anonymous lambdas because they're super short and the code speaks for itself. I get that you're emphasizing that in one case the presence of a key is checked for and in another the property is unconditionally set but I think it would be clear enough without explicit function names. /My two cents :|

Reply:
`widenAncestors` wasn't going to play nice with the next commit, which needed to set same style values, but unconditionally, and not on ancestors, rather, on a specific element.

I hear ya though. I tried to keep the new bits ultra-dead simple, with declarative naming so it's easy to see "it's doing this to the ancestors and it's doing that to the anchor".